include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -434,10 +434,11 @@ struct __parallel_scan_submitter<_CustomName, __internal::__optional_kernel_name
                                                                              sizeof(_Type), __wgroup_size);
 
 #if _ONEDPL_COMPILE_KERNEL
-        auto __kernel_1 =
-            __internal::__kernel_compiler<_LocalScanKernel>::__compile_kernel(::std::forward<_ExecutionPolicy>(__exec));
-        auto __kernel_2 =
-            __internal::__kernel_compiler<_GroupScanKernel>::__compile_kernel(::std::forward<_ExecutionPolicy>(__exec));
+        //Actually there is one kernel_bundle for the all kernels of the pattern.
+        auto __kernels = __internal::__kernel_compiler<_LocalScanKernel, _GroupScanKernel>::__compile(
+            ::std::forward<_ExecutionPolicy>(__exec));
```

Comment:
It's still dangerous to forward policy keeping in mind that we use it later. But as we discussed there lots of such places. We need to either do global refactoring for that or reconsider the approach. Please create the separate issue with that regard.